### PR TITLE
Add kv indexing and iterator functions for staking

### PR DESCRIPTION
**Pull request by dongsam**
- add farming module account on genesis with expected keeper
- add `StakingByFarmer`, `StakingByStakingCoinDenom` indexing for stakings
- cherrypick some functions from https://github.com/tendermint/farming/pull/15

### CODE REVIEW

File: x/farming/keeper/keeper.go [hunk 1]
```diff
@@ -118,3 +120,13 @@ func (k Keeper) UnmarshalPlan(bz []byte) (types.PlanI, error) {
 
 // GetCodec return codec.Codec object used by the keeper
 func (k Keeper) GetCodec() codec.BinaryCodec { return k.cdc }
+
+// GetStakingCreationFeePool returns module account for collecting Staking Creation Fee
+func (k Keeper) GetStakingCreationFeePool(ctx sdk.Context) authtypes.ModuleAccountI { // nolint:interfacer
+	return k.accountKeeper.GetModuleAccount(ctx, types.ModuleName)
+}
+
+// GetStakingCreationFeePool returns module account for collecting Staking Creation Fee
```
Comment (hallazzang) [inline x/farming/keeper/keeper.go:129]: Need to update comment

File: x/farming/types/expected_keepers.go [hunk 2]
```diff
@@ -25,6 +25,8 @@ type BankKeeper interface {
 type AccountKeeper interface {
 	GetAccount(ctx sdk.Context, addr sdk.AccAddress) authtypes.AccountI
 	GetModuleAddress(name string) sdk.AccAddress
+	GetModuleAccount(ctx sdk.Context, moduleName string) authtypes.ModuleAccountI
+	SetModuleAccount(sdk.Context, authtypes.ModuleAccountI)
```
Comment (hallazzang) [inline x/farming/types/expected_keepers.go:29]: We can remove this line

File: x/farming/types/keys.go [hunk 3]
```diff
@@ -84,13 +87,18 @@ func GetStakingByFarmerAddrIndexKey(farmerAcc sdk.AccAddress) []byte {
...
 
-//// GetStakingByStakingCoinDenomIdIndexKey returns key for the staking index by the staking coin denomination
-//func GetStakingByStakingCoinDenomIdIndexKey(denom string, id uint64) []byte {
-//	return append(StakingByFarmerAddrIndexKeyPrefix, MustLengthPrefixString(denom)...)
-//}
+// GetStakingByStakingCoinDenomIdIndexKey returns key for the staking index by the staking coin denomination
+func GetStakingByStakingCoinDenomIdIndexKey(denom string, id uint64) []byte {
+	idBytes := make([]byte, 8)
+	binary.BigEndian.PutUint64(idBytes, id)
+	fmt.Println(MustLengthPrefixString(denom))
```
Comment (hallazzang) [inline x/farming/types/keys.go:97]: we need to remove `fmt.Println`s

File: x/farming/types/staking.go [hunk 4]
```diff
@@ -1,13 +1,35 @@
...
 
-import sdk "github.com/cosmos/cosmos-sdk/types"
+import (
+	"encoding/binary"
 
-func (staking Staking) GetFarmerAddress() sdk.AccAddress {
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func (staking Staking) FarmerAddress() sdk.AccAddress {
```
Comment (hallazzang) [inline x/farming/types/staking.go:9]: `FarmerAddress` -> `GetFarmerAddress`

**Review by hallazzang — CHANGES_REQUESTED** [review]

File: x/farming/types/expected_keepers.go (hunk 2 above)
Comment (dongsam) [reply x/farming/types/expected_keepers.go:29]: I added usages of `SetModuleAccount` on `initGenesis`

**Pull request merged**